nexus/src/app/background/tasks/fm_sitrep_load.rs

```diff
@@ -0,0 +1,179 @@
+                    return Status::Error(format!(
+                        "previously loaded sitrep {id:?} (v{version}), \
+                         but now no sitreps exist",
+                    ));
+                }
+                None => {
+                    // Had no sitrep; still have no sitrep.
+                    return Status::NoSitrep;
+                }
+            },
+            Err(err) => {
+                let err = InlineErrorChain::new(&err);
+                warn!(
+                    log,
+                    "failed to read current sitrep version";
+                    &err
+                );
+                return Status::Error(format!(
+                    "failed to read current sitrep version: {err}"
+                ));
+            }
+        };
+
+        // Have we already loaded this sitrep?
+        match old {
+            Some(version) if version.id == current_version.id => {
+                debug!(log, "current sitrep has not changed");
+                return Status::Loaded { version, time_loaded };
+            }
+            Some(SitrepVersion { version, id, .. })
+                if version == current_version.version
+                    && id != current_version.id =>
+            {
+                // Well, this is weird! Entries in the `sitrep_version` table
+                // should not change IDs once they are created, that seems like
+                // a bug. Nonetheless, we will load the new UUID, but we should
+                // say something about this, as it's a bit odd.
+                warn!(
+                    log,
+                    "sitrep ID associated with the current version in the \
+                     database has changed; this is not supposed to happen!";
+                     "current_id" => ?current_version.id,
+                );
+            }
+            _ => (),
+        }
+
+        let sitrep = match self
+            .datastore
+            .fm_sitrep_read(opctx, current_version.id)
+            .await
+        {
+            Ok(sitrep) => sitrep,
+            Err(err) => {
+                let err = InlineErrorChain::new(&err);
+                error!(
+                    log,
+                    "failed to load current sitrep";
+                    "current_id" => ?current_version.id,
+                    "current_version" => ?current_version.version,
+                    &err
+                );
+                return Status::Error(format!(
+                    "failed to read current sitrep {:?} (v{}): {err}",
+                    current_version.id, current_version.version
+                ));
+            }
+        };
+
+        let sitrep = Arc::new((current_version.clone(), sitrep));
+        self.tx.send_modify(|s| {
+            *s = Some(sitrep);
+        });
+
+        Status::Loaded { version: current_version, time_loaded }
+    }
+}
+
+#[cfg(test)]
+mod test {
+    // TODO
```

Comment:
Seems like we should be able to write a test in this PR:

- Load no sitrep
- Insert a sitrep
- Load the sitrep
- Load the sitrep again

Reply:
done in 8522c68e26112e237b60a94827f793b5c4d55139